### ISSUE TITLE
Cleanup too aggressive

### ISSUE THREAD

**Issue by xoferz** [kotope/aio_energy_management#9]
When new Nordpool prices are published and starting_today is false the hours are removed from today even if they have not yet arrived.

For instance, I have a following sensor:
```
    - nordpool_entity: sensor.nordpool
      unique_id: nordpool_cheapest_2h_nonconsecutive
      name: "Cheapest 2h non-consecutive"
      first_hour: 0
      last_hour: 22
      starting_today: false
      number_of_hours: 2
      sequential: False
      failsafe_starting_hour: 22  
```
One of the hours was today - 25/09 at 15:00-16:00
This hour was removed from calendar when the prices for tomorrow, 26/09 were published.
![Screenshot 2024-09-25 at 14 40 30](https://github.com/user-attachments/assets/af861fb1-0ebc-4a98-818e-eb0f7804c5fb)
(it is 14:50 now -- 15:00 has still not arrived and it has vanished)
Only events in the past should be cleaned up from calendar. I would prefer to even keep past events for today for visuals.

**Comment by kotope** [comment]
Is this with the latest 0.2.0 release? I did make some changes of the expiration logic since it failed with entso-e and certain configuration. So if with latest release I might have done some regression accidentally.

Will look into this.

**Comment by xoferz** [comment]
0.1.1

**Comment by kotope** [comment]
> 0.1.1

Ok, could you try with latest 0.2.0? This issue is most probably fixed already. Just tried to reproduce this using unit testing and couldn't get this to happen.
Will try the same unittest with 0.1.1 to see if it fails on that release.

**Comment by kotope** [comment]
Couldn't reproduce this issue either with 0.1.1. 

Would you mind sharing some more info? I'd like to know the timezone you're in and the Nord Pool regional data used.

**Comment by kotope** [comment]
<img width="991" alt="Screenshot 2024-09-25 at 15 53 22" src="https://github.com/user-attachments/assets/82f4649d-333d-46e2-855c-72b5f2c03b4b">

I've got kind of similar setup on my dev env currently. "My expensive hours" is still coming up later today and already tomorrow hours are ready for tomorrow, no cleanup is made for that entity. This is with latest 0.2.0 though.

**Comment by xoferz** [comment]
Are they "starting_today: false"? I do not really understand the implications of true/false here anyway :-)

**Comment by kotope** [comment]
This should now be fixed on 0.2.1 assuming that the root cause for this is the reboot of Home Assistant or reloading integration.

If you still experience data loss, please re-open the issue and I'll keep on digging.

Thank you @xoferz for your bug report!

**Comment by xoferz** [comment]
Yes, HA reload might have well been a factor. Will test.
